applications/leaderboard/Leaderboard/Program.cs

```diff
@@ -273,14 +271,13 @@ static bool WasTransactionAborted(AggregateException exception)
             $"Data Source=projects/{projectId}/instances/{instanceId}"
             + $"/databases/{databaseId}";
 
-            using (TransactionScope scope = new TransactionScope(
-               TransactionScopeAsyncFlowOption.Enabled))
+            // Insert 4 score records into the Scores table for each player
+            // in the Players table.
+            using (var connection = new SpannerConnection(connectionString))
             {
-                // Insert 4 score records into the Scores table for each player
-                // in the Players table.
-                using (var connection = new SpannerConnection(connectionString))
+                await connection.OpenAsync();
+                await connection.RunWithRetriableTransactionAsync(async (transaction) =>
                 {
-                    await connection.OpenAsync();
                     Random r = new Random();
                     bool playerRecordsFound = false;
                     SpannerBatchCommand cmdBatch =
```

Comment:
nit: for line 295 change the for as follows, is more idiomatic

`for (int i =0; i < 4; i++)`

Reply:
Done.